test/mpi/buffer.cc

```diff
@@ -85,6 +98,25 @@ static inline bool is_equal(T a, T b, T epsilon)
     return fabs(a - b) <= ((fabs(a) < fabs(b) ? fabs(b) : fabs(a)) * epsilon);
 }
 
+static inline bool is_equal_complex(float _Complex a, float _Complex b,
+                                    double epsilon)
+{
+    return cabsf(a - b) <=
+           ((cabsf(a) < cabsf(b) ? cabsf(b) : cabsf(a)) * epsilon);
```

Comment:
not sure it's good way to compare complex numbers, i would better do something like this

return is_equal(a.real, b.real) && is_equal(a.im, b.im)

Reply:
Ok, I made the changes